src/Tribe/Views/V2/View.php

```diff
@@ -1568,4 +1569,44 @@ protected function filter_display_events_bar( $display ) {
 
 		return $display;
 	}
+
+	/**
+	 * Returns the iCal data we're sending to the view.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	protected function get_ical_data() {
+		/**
+		 * A filter to control whether the "iCal Import" link shows up or not.
+		 *
+		 * @since unknown
+		 *
+		 * @param boolean $show Whether to show the "iCal Import" link; defaults to true.
+		 */
+		$display_ical = apply_filters( 'tribe_events_list_show_ical_link', true );
+
+		/**
+		 * Allow for customization of the iCal export link "Export Events" text.
+		 *
+		 * @since unknown
+		 *
+		 * @param string $text The default link text, which is "Export Events".
+		 */
+		$link_text  = apply_filters( 'tribe_events_ical_export_text', __( 'Export Events', 'the-events-calendar' ) );
+
+		$link_title = __( 'Use this to share calendar data with Google Calendar, Apple iCal and other compatible apps', 'the-events-calendar' );
+
+		$ical_data = [
+			'display_link' => $display_ical,
+			'link'         => (object) [
+				'url'    => esc_url( tribe_get_ical_link() ),
+				'anchor' => $link_text,
+				'title'  => $link_title,
+			],
+		];
+
+		return (object) $ical_data;
```

Comment:
any specific reason why you're typecasting the data to objects here? not sure of the pros and cons here from a BE perspective.

Reply:
I would preferer it to be an array, makes it easier to work with `Arr::get()`

Reply:
Made it this way so we can be consistent with the rest (thumbnails, tickets) and use, for example: `$ical->link->title` in the template.

Reply:
@bordoni this info goes to the template and ideally we could use it like we do with the post thumbs and the rest -> `$ical->link->title`. Do you want to use the array utils functions in the templates?

Reply:
Ohh I see, this information goes into the `Event_Model`? For `tribe_get_event()`.

Reply:
It's view based, not related to the event: https://github.com/moderntribe/the-events-calendar/pull/2860/files#diff-828c168861c5b8f4cb8331c20c65d55aR1102